Intermediate certificate documentation (#3254)
EventStore/EventStore: docs/security/configuration.md

| YAML                 | `CertificateThumbprint` |
| Environment variable | `EVENTSTORE_CERTIFICATE_THUMBPRINT` |

### Intermediate CA certificates

> Intermediate certificates are supported by loading them from a PEM or PKCS #12 bundle. The node's certificate should be first in the bundle, followed by the intermediates (intermediates can be in any order but it would be good to keep it from leaf to root as per the usual convention). The certificate chain is validated on startup with the node's own certificate.

Reviewer: ```suggestion
Intermediate certificates are supported by loading them from a [PEM](https://www.rfc-editor.org/rfc/rfc1422) or [PKCS #12](https://datatracker.ietf.org/doc/html/rfc7292) bundle. The node's certificate should be first in the bundle, followed by the intermediates (intermediates can be in any order but it would be good to keep it from leaf to root as per the usual convention). The certificate chain is validated on startup with the node's own certificate.
```

Reviewer: **QUESTION:** Do we have some recommendations on when to use and when not to use intermediate CA certificates?

Could we also show the example bundle structure or add a link to how to configure such a bundle?

Author: It's always better to use one or more intermediate certificates as this is more secure, since the root CA is not directly used. People who deal with PKI should know this :)

We could show an example once the `es-gencert-cli` is updated, but not sure if those certs should be used in production and if it's wise to show it to people.

Reviewer: I think that the audience trying to configure those certificates will not always be people dealing with PKI daily, so IMHO, it's always worth adding more context.

Regarding samples, I was more referring to somehow visualising or explaining the structure explained in the description.

Author: Think this info is more for people who are already using intermediate certificates and want to know if it's supported and how. For others, things regarding certificates will just stay as-is.  

Don't think we are in the business of telling how to do PKI, in the mongodb docs they are referring to [prior knowledge](https://docs.mongodb.com/manual/tutorial/configure-ssl/#prerequisites)

Not sure if I'm the right person to make something visual for the website/docs 😬

Additionally, in order to improve performance, the server will also try to bypass intermediate certificate downloads, when they are available on the system in the appropriate locations:

> #### Steps on Linux:

Reviewer 2: ```suggestion
#### Steps on Linux:

The following script assumes EventStoreDB is running under the `eventstore` account

```

```
sudo su eventstore --shell /bin/bash
dotnet tool install --global dotnet-certificate-tool
 ~/.dotnet/tools/certificate-tool add --file /path/to/intermediate.crt
```

#### Steps on Windows: 

> Right click on the `intermediate certificate` -> `Install` (let windows choose the appropriate location - it should go under `Intermediate Certification Authorities` -> `Certificates` in the current user's certificate store.

Reviewer 2: ```suggestion
In the Certificate store under which EVentStoreDB is running or the `Local Computer` store, right click on the `intermediate certificate` -> `Install` (let windows choose the appropriate location - it should go under `Intermediate Certification Authorities` -> `Certificates` 
```

Reviewer: @pvanbuijtene question. Cannot we use the `Import-Certificate` PowerShell command for that? See more: https://github.com/EventStore/EventStore-Client-Dotnet/blob/master/samples/secure-with-tls/create-certs.ps1#L5

Author: Good point, will test it

Author: Successfully tested the PowerShell commands and updated the docs accordingly.
